File: Python/Program Insertion Sort.py
print("Pengurutan Data dengan Algoritma Insertion Sort")
print("visit us hobingoding.com")
print()
data = []
# Input Banyak Data
banyak_data = int(input("Input Banyak Data : "))

# Input Data
print("Input Data :")
for i in range(banyak_data):
    data.append(int(input()))

# Algoritma Insertion Sort
for i in range(banyak_data):
    temp = data[i]
    j = i - 1
    while(j >= 0 and temp < data[j]):
        data[j+1] = data[j]
        j -= 1
    data[j + 1] = temp

# Hasil Pengurutan Data
print("Hasil Pengurutan Data :", data)

File: C/Program Insertion Sort.c
#include <stdio.h>

int main() {
    int banyak_data, i, j;

    printf("Pengurutan Data dengan Algoritma Insertion Sort\n");
    printf("visit us hobingoding.com\n\n");
    // Input Banyak Data
    printf("Input Banyak Data : ");
    scanf("%d", &banyak_data);
    int data[banyak_data];

    // Input Data
    printf("Input Data : ");
    for(i = 0; i < banyak_data; i++) {
        scanf("%d", &data[i]);
    }

    // Algoritma Insertion Sort
    for(i = 0; i < banyak_data; i++) {
        for(j = i; j > 0 && data[j] < data[j-1]; j--) {
           int temp = data[j-1];
           data[j-1] = data[j];
           data[j] = temp;
        }
    }

    // Hasil Pengurutan Data
    printf("Hasil Pengurutan Data : ");
    for (i = 0; i < banyak_data; i++) {
        printf("%d ", data[i]);
    }
    return 0;
}
